File: AppTool/src/main/java/com/luoye/apptool/viewbinding/activity/BaseViewFragmentActivity.java
package com.luoye.apptool.viewbinding.activity;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewbinding.ViewBinding;

import com.luoye.apptool.tool.ActivityTool;
import com.luoye.apptool.utils.ViewBindingUtil;

public abstract class BaseViewFragmentActivity<T extends ViewBinding> extends AppCompatActivity {

    protected static String TAG = "---";
    protected Context context;
    protected T binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ViewBindingUtil.create(getClass(), LayoutInflater.from(this));
        setContentView(binding.getRoot());
        context = this;
        ActivityTool.getInstance().addActivity(this);
        initActivity();
    }

    protected abstract void initActivity();


    @Override
    protected void onDestroy() {
        super.onDestroy();
        ActivityTool.getInstance().removeActivity(this);
    }
}

File: ktBase/src/main/java/com/luoye/ktbase/binding/viewbinding/activity/BaseViewFragmentActivity.kt
package com.luoye.ktbase.binding.viewbinding.activity

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import androidx.fragment.app.FragmentActivity
import androidx.viewbinding.ViewBinding
import com.luoye.ktbase.binding.ViewBindingUtil
import com.luoye.ktbase.tool.ActivityTool

/**
 *
 * Created by: luoye
 * Time: 2022/11/30
 * user:
 *
 */
abstract class BaseViewFragmentActivity<V : ViewBinding> : FragmentActivity() {

    protected var TAG = "---"
    protected lateinit var context: Context
    protected lateinit var binding: V

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ViewBindingUtil.create(javaClass, LayoutInflater.from(this))
        setContentView(binding.getRoot())
        context = this
        ActivityTool.instance.addActivity(this)
        initActivity()
    }

    protected abstract fun initActivity()


    override fun onDestroy() {
        super.onDestroy()
        ActivityTool.instance.removeActivity(this)
    }

}